### PR TITLE
feat: Add member avatar endpoint with related methods

### PR DESCRIPTION
**Pull request by sudojunior**
## Added

- `ResolvedMember#avatarURL()`
- `ResolvedMember#dynamicAvatarURL(format?: ImageFormat, size?: number)`
- `ResolvedMemberData.avatar?`

---

- `ResolvedMember` and `Member` (as child of the former) now take a `guildID` argument **after** the `creator` argument.
  > Existence of `guildID` is guaranteed if `Member` or `ResolvedMember` is instantiated.

## Aside

* `ResolvedMember#id` *can* also be converted into a getter since it's from `userData`.
* `ResolvedMember#is_pending` is not a property of the [GuildMember](https://discord.dev/docs/resources/guild#guild-member-object) structure (partial or not).

### CODE REVIEW

**Comment by Snazzah** [comment]
Is this still active? Not sure if these props are exposed in resolved roles. I do have a server to test this in though.

**Comment by sudojunior** [comment]
Main hitch continues to be that the guild ID isn't passed through to the role structure, therefore making it likely to be a breaking change.

> If it cannot progress further at this time, please close as stale.

**Comment by Snazzah** [comment]
You could pass the guild ID into the constructor of the Member and ResolvedMember classes, that should work

**Comment by sudojunior** [comment]
Closing as my environment can no longer progress with routine testing... _error is run on this PR, but does not happen on `main`_.
```
$ npx mocha -r ts-node/register --extension ts 'test/**/*.ts'

TypeError: Class extends value undefined is not a constructor or null
    at Object.<anonymous> (E:\git\slash-create\src\structures\member.ts:8:29)
```

File: src/structures/resolvedMember.ts [hunk 1]
```diff
@@ -54,4 +58,21 @@ export class ResolvedMember {
...
+  dynamicAvatarURL(format?: ImageFormat, size?: number) {
+    if(!this.avatar) return this.user.dynamicAvatarURL(format, size);
+    if(!format || !ImageFormats.includes(format.toLowerCase())) {
+      format = this.avatar.startsWith('a_') ? 'gif' : this._creator.options.defaultImageFormat;
+    }
+    if(!size || size < ImageSizeBoundaries.MINIMUM || size > ImageSizeBoundaries.MAXIMUM) {
+      size = this._creator.options.defaultImageSize;
+    }
+
+    return `${CDN_URL}${Endpoints.GUILD_MEMBER_AVATAR(null /* guildID - not available */, this.id, this.avatar)}.${format}?size=${size}`;
```
Comment (Snazzah) [inline src/structures/resolvedMember.ts:76]: I'm pretty sure this is why tests are failing, you assign null to a string parameter.

**Pull request merged**